### ISSUE TITLE
Improve search functionality

### ISSUE THREAD

**Issue by AnaLoznianu** [oceanprotocol/market#1635]
add price filters/sorts
```
"price": {
      "value" : 33,
      "tokenAddress": "0x0....",
      "tokenSymbol": "ocean"
}
``` 
Come up with better UI for sort/filters . Add more criteria . Like top tags,  maybe a text box to search for a tag
What more sorting feature can we add?

**Comment by AnaLoznianu** [comment]
Another bug that should be fixed when improving the search functionality.


https://user-images.githubusercontent.com/84312885/186369061-a09060b7-8fba-47fc-801a-8be5e0eea48a.mov

**Comment by jamiehewitt15** [comment]
We can filter via any of the fields that aqua returns but I don't think there are many more that are useful as sorting or filtering criteria. 

![Aqua result - 2022-08-24_15-18](https://user-images.githubusercontent.com/20739535/186416790-0988979a-d8db-443e-a47c-1b442e53a6de.png)

**Comment by andreaarmanni** [comment]
Premise: as we’re working to grow DCV on Ocean Market via data challenges, it would be useful to have more social aspects added to it - whether this means improving the way profiles are displayed, or simply augmenting filtering/search functionality. These are some of the changes I’d like to propose:

- [ ] Change the format of the URL from     /search?tags=bounty&sort=_score&sortOrder=desc    to     /tags/bounty
_Reason: with this change it will be easier for everyone to find and share all datasets about a specific topic. Whether the tag is a company, an industry, or an initiative (like data challenges) it would make things easier to remember/share and filter by tag/URL_
Importance: High

- [ ] Allow users to give a name/username to their profile and, when the name/username is available, change the URL to reflect such change so it looks like /profile/username
_Reason: currently the only way to have a readable username is via ENS integration which not many people have. As every marketplace, the social aspect is crucial and this change would help with making a profile more sharable/human. If improved overtime, a good profile page could lead to less people forking ocean market because they can leverage on their “private” area._
Importance: High

- [ ] Allow users to add a profile photo, description, twitter & Github handle to their profile
_Reason: Similarly to the reason above, this would help towards better customisation. It would also help towards establishing a connection between data publisher and buyer since they can share their social accounts if they want to. If improved overtime, it could lead to less people forking ocean market because they can leverage on their “private” area._
Importance: Medium

- [ ] Add a “Popular tags” section at the top of the home page to showcase what are the most searched/used tags on the market. 
Reason: This hopefully is the first step towards incentivising users to pick consistent tags rather than creating random ones. The idea is to leverage on tags to create a low effort, high impact experience for navigating the market.
Importance: Medium

- [ ] Below each data asset add a section to showcase all data assets that are somehow related to the ones into the profile.
Reason: similarly to how Google handles backlinks, or Amazon handles suggestions, it would be interesting to find a way to showcase datasets that are similar to the one searched for (an initial suggestion could come from filtering per tags and sales). As the market grows, this could help users find datasets relevant to their case.
Importance: Low

**Comment by kremalicious** [comment]
This is quickly getting out of scope. We are not going to create our own database holding custom usernames. As Ocean Protocol, we are not going to solve decentralized user data. ENS right now is the only reliable and widely supported way of doing user profile data. So that's what we support. If industry comes up with something better, we switch to it.

>  Allow users to give a name/username to their profile

How would that work decentralized without a database? And if we have a database how does this work across multiple marketplaces?

> Allow users to add a profile photo, description, twitter & Github handle to their profile

Same question.

All the other points are great points and can be done with our existing search functionality based on ElasticSearch.

**Comment by mihaisc** [comment]
I'll create relevant issues , because they are not relevant for this one

**Comment by andreaarmanni** [comment]
Thanks for the feedback and for opening new issues. Unelss I missed it, I believe the first point remains undiscussed: 

Would it be possible to change the format of the URL from `/search?tags=bounty&sort=_score&sortOrder=desc` to `/tags/bounty` ? The reason is to leverage on tags to make it easier for users to find/remember/share all data assets about a specific topic.

**Comment by mihaisc** [comment]
Not that easy to do and impacts multiple components/functionalities. Also don't really see the benefit. If you remember the tag you can just search for it.
For example amazon url for noise canceling headphones : https://www.amazon.co.uk/b/?_encoding=UTF8&node=310193011&bbn=560798&ref_=Oct_d_odnav_d_4085731_2&pd_rd_w=5Sy8Q&content-id=amzn1.sym.d191d14d-5ea3-4792-ae6c-e1de8a1c8780&pf_rd_p=d191d14d-5ea3-4792-ae6c-e1de8a1c8780&pf_rd_r=9H6X2VYGHMC6PHGYK7GX&pd_rd_wg=ysFMU&pd_rd_r=47f5e12a-9c7e-419e-99b2-ba77a17fe1d9

TLDR: too much effort for not that much benefit 😄
